File: packages/renderer-vue/playground/measurements/MeasurementAreaSumNode.ts
import { defineNode, NodeInterface } from "@baklavajs/core";
import { setType } from "@baklavajs/interface-types";
import { Measurement } from "../models/Measurement";
import { numberType, measurementListType } from "../interfaceTypes";

export default defineNode({
    type: "MeasurementAreaSumNode",
    title: "Area Sum",
    inputs: {
        measurements: () => new NodeInterface<Measurement[]>("Measurements", []).use(setType, measurementListType),
    },
    outputs: {
        areaSum: () => new NodeInterface<number>("Area Sum", 0).use(setType, numberType),
    },
    calculate(inputs) {
        // Sum up all the areas from the measurements array
        let areaSum = 0;

        // inputs.measurements should be an array of measurements
        if (Array.isArray(inputs.measurements)) {
            for (const measurement of inputs.measurements) {
                if (measurement && typeof measurement.area === 'number') {
                    areaSum += measurement.area;
                }
            }
        }

        return { areaSum };
    },
});

File: BaklavaDependencyEngine/BaklavaDependencyEngine.Core/ExampleNodes/MeasurementAreaSumNode.cs
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using BaklavaDependencyEngine.Core.Models;

namespace BaklavaDependencyEngine.Core.ExampleNodes
{
    public class MeasurementAreaSumNode : Node
    {
        public MeasurementAreaSumNode() : base("MeasurementAreaSumNode", "Area Sum")
        {
        }

        protected override void InitializeInterfaces()
        {
            AddInput("measurements", new NodeInterface("Measurements", new List<Measurement>()));
            AddOutput("areaSum", new NodeInterface("Area Sum", 0.0, false));
        }

        public override async Task<Dictionary<string, object>> Calculate(
            Dictionary<string, object> inputs,
            CalculationContext context)
        {
            var measurements = inputs["measurements"] as List<Measurement> ?? new List<Measurement>();

            double areaSum = 0.0;
            foreach (var measurement in measurements.Where(m => m != null))
            {
                areaSum += measurement.Area;
            }

            return await Task.FromResult(new Dictionary<string, object>
            {
                ["areaSum"] = areaSum
            });
        }
    }
}